services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/streaming/StreamingPropertiesRequestProcessor.java

```diff
@@ -0,0 +1,95 @@
+import edu.unc.lib.boxc.model.api.services.RepositoryObjectFactory;
+import edu.unc.lib.boxc.model.fcrepo.ids.PIDs;
+import edu.unc.lib.boxc.operations.jms.streaming.StreamingPropertiesRequest;
+import edu.unc.lib.boxc.operations.jms.streaming.StreamingPropertiesRequestSerializationHelper;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.tika.utils.StringUtils;
+
+import java.io.IOException;
+
+import static edu.unc.lib.boxc.operations.jms.streaming.StreamingPropertiesRequest.VALID_FOLDERS;
+import static org.apache.commons.io.FilenameUtils.getName;
+import static org.apache.commons.io.FilenameUtils.removeExtension;
+
+/**
+ * Processing requests to edit streaming properties on a FileObject
+ */
+public class StreamingPropertiesRequestProcessor implements Processor {
+    private RepositoryObjectLoader repositoryObjectLoader;
+    private RepositoryObjectFactory repositoryObjectFactory;
+    private AccessControlService aclService;
+
+    @Override
+    public void process(Exchange exchange) throws IOException {
+        var in = exchange.getIn();
+        var request = StreamingPropertiesRequestSerializationHelper.toRequest(in.getBody(String.class));
+        var agent = request.getAgent();
+        var pid = PIDs.get(request.getFilePidString());
+
+        aclService.assertHasAccess("User does not have permission to set streaming properties",
+                pid, agent.getPrincipals(), Permission.ingest);
+
+        if (isValid(request, pid)) {
+            var file = repositoryObjectLoader.getFileObject(pid);
+            repositoryObjectFactory.createExclusiveRelationship(
+                    file, Cdr.streamingHost, request.getHost());
+            repositoryObjectFactory.createExclusiveRelationship(
+                    file, Cdr.streamingFile, formatFilename(request.getFilename()));
+            repositoryObjectFactory.createExclusiveRelationship(
+                    file, Cdr.streamingFolder, request.getFolder());
+        } else {
+            throw new IllegalArgumentException("Both a filename and streaming folder are required.");
+        }
+    }
+
+    private boolean isValid(StreamingPropertiesRequest request, PID pid) {
+        var folder = request.getFolder();
+        if (StringUtils.isBlank(request.getFilename()) || StringUtils.isBlank(folder)) {
+            return false;
+        }
+
+        if (!VALID_FOLDERS.contains(folder)) {
+            return false;
+        }
+
+        try {
+            repositoryObjectLoader.getFileObject(pid);
+        } catch (ObjectTypeMismatchException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Converts a filename to appropriate format for streaming
+     * A filename like a/b/banjo_recording.mp3 would transform to banjo_recording-playlist.m3u8
+     * @param filename filename of the file object
+     * @return formatted string
+     */
+    private String formatFilename(String filename) {
```

Comment:
i don't have a good sense of what the existing workflow for setting streaming properties is like. Will the user setting these properties only know about the "banjo_recording.mp3" filename, or will they know about the m3u8 file name? If its just a matter of what's currently stored in CDM being filenames rather than playlists, we could convert that as part of the migration, but if its going to be an ongoing situation then it makes sense to do it here

Reply:
with the current usage for CDM, it just takes what's in CDM as the filename. so you could do it as part of the migration if that's the only source for boxc?

Reply:
What i'm wondering is if for setting these properties on File objecst in the admin ui, if the user is going to know the .mp3 file or the .m3u8 file when they input the value. But perhaps it doesn't matter too much since this method should force it into m3u8 format either way (is that format used for stream audio and video?). Although one issue is that if the user does input `banjo_recording-playlist.m3u8` then this method will turn it into `banjo_recording-playlist-playlist.m3u8`, so we should probably address that